### ISSUE TITLE
Two PV systems with storage on separate load centers fails with storage not found error

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4921]
This is followup from #4015 .  When trying to test the output variable fix (same name but different units) the test file failed with a false object not found error:

```
   ** Severe  ** GetPowerManagerInput: ElectricLoadCenter:Distribution="PV ARRAY LOAD CENTER", invalid entry.
   **   ~~~   ** Invalid Electrical Storage Object Name = KIBAM
   **   ~~~   ** Electrical storage object was not found; Must have and be valid when Buss Type="DirectCurrentWithInverterDCStorage".
```

This issue has three parts:
- [ ] Fix the fatal error - unless I misunderstand, it should be possible to have these two load centers present with PV and storage.
- [ ] Once the test file runs, then confirm that both of these output variables appear in the rdd and flow through to the csv.

```
Output:Variable,*,Electric Storage Charge State,hourly; !- HVAC Average [J]
Output:Variable,*,Electric Storage Charge State,hourly; !- HVAC Average [Ah]
```
- [ ] Consider whether one of these output variables should be changed - either different name, or consistent units.

**Comment by Nigusse** [comment]
correcting get input index problem eliminated the fatal error ( the defect file runs successfully).  I need to check the output variables though.  The two storage charge variables are not reported unless there is an EMS defined in the idf.  Why do we have this restriction?

if ( AnyEnergyManagementSystemInModel ) {
    if ( ElecStorage( StorNum ).StorageModelMode == SimpleBucketStorage ) {
        SetupEMSInternalVariable( "Electrical Storage Maximum Capacity", ElecStorage( StorNum ).Name, "[J]", ElecStorage( StorNum ).MaxEnergyCapacity );
    } else if ( ElecStorage( StorNum ).StorageModelMode == KiBaMBattery ) {
        SetupEMSInternalVariable( "Electrical Storage Maximum Capacity", ElecStorage( StorNum ).Name, "[Ah]", ElecStorage( StorNum ).MaxAhCapacity );
            }
}  

I will remove the EMS requirement for reporting purpose. However the units for this variables are not equivalent.  And the Simple represents a generic storage and the other is manufacturers specific, I think, and Battery manufacturers usually report capacity in "Ah".  So I would rather keep both variables but we may modify the names.  Suggested name changes:

 Output:Variable,_,Simple Electric Storage Charge State,hourly; !- HVAC Average [J]
 Output:Variable,_,Battery Electric Storage Charge State,hourly; !- HVAC Average [Ah]

@mjwitte @Myoldmopar @EnergyArchmage Any thought?

**Comment by Nigusse** [comment]
For consistency with other report variable of the electric storage models, I propose the following alternative variable names:

Output:Variable,,Electric Storage Simple Charge State,hourly; !- HVAC Average [J]
Output:Variable,,Electric Storage Battery Charge State,hourly; !- HVAC Average [Ah]

**Comment by mjwitte** [comment]
@Nigusse The EMS requirement is correct for SetupEMSInternalVariable.  The code pasted in the above comment looks correct.

**Comment by Nigusse** [comment]
@mjwitte OK.  I see now.  They should remain as is. Should we modify the variables name as suggested above?

**Comment by mjwitte** [comment]
@Nigusse I don't have a strong opinion either way on the output variables.  If they are changed, I prefer the second proposal Electric Storage Simple/Battery Charge State.  But you should wait to hear from @EnergyArchmage to avoid conflicts with the larger PV/storage work that is ongoing.

**Comment by EnergyArchmage** [comment]
I do have this issue in my list to include in the larger project.  But I have been struggling with what to do about the units here.  Amp hours is so basic to the real world battery that it seems hard to make it change to be Joules.  I think the slight change in output variable text slight is a good approach.  I also thought of eventually making both Joules and Amp-Hours flavors of reports available for both types of batteries.  

So maybe something like this?  would work for both types of batteries. 
Electric Storage Charge State [J]
Electric Storage Charge Level [Ah]

**Comment by Nigusse** [comment]
The Simple/Battery is an indicator of the Storage Type, that is why I suggested. Either way I am fine.

**Comment by mjwitte** [comment]
I like having both Ah and J outputs for both types of storage.

**Comment by EnergyArchmage** [comment]
Am finding it difficult to get both types of output for both types of storage.  There is no voltage defined for the simple storage.  And for the kinetic model it isn't clear what voltage should be used to convert.  I am starting to like @Nigusse suggestion of just splitting the output names.

**Comment by Nigusse** [comment]
I agree. There are a number of factors that affect the operating voltage of the battery, for instance the voltage you draw depends on the balance between charging cycle per year and the life of battery selected. I think for now I second to keep a separate variable name.

**Comment by EnergyArchmage** [comment]
Closed via #5437
